packages/loader/container-loader/src/container.ts

```diff
@@ -1067,16 +1067,17 @@ export class Container
 		}
 	}
 
-	public closeAndGetPendingLocalState(): string {
+	public async closeAndGetPendingLocalState(): Promise<string> {
 		// runtime matches pending ops to successful ones by clientId and client seq num, so we need to close the
 		// container at the same time we get pending state, otherwise this container could reconnect and resubmit with
 		// a new clientId and a future container using stale pending state without the new clientId would resubmit them
-		const pendingState = this.getPendingLocalState();
+		this.disconnect(); // TODO investigate side effects
+		const pendingState = await this.getPendingLocalState({ shutdownBlobUpload: true });
 		this.close();
 		return pendingState;
 	}
 
-	public getPendingLocalState(): string {
+	public async getPendingLocalState(props?: { shutdownBlobUpload?: boolean }): Promise<string> {
```

Comment:
this should not take the props object, we don't want this behavior will this method is called directly. you may need to refactor the relationship between the two methods.

Reply:
Maybe a flag at class level?

Reply:
no. just create a private method getPendingLocalStateCore or soemthing, that both getPendingLocalState and closeAndGetPendingLocalState can call with props. i just don't want them on the public interface

Reply:
it might even be fine for them to remain here as optional, and not have them on the interface, so only the class itself can use them